server/controllers/reaction.js

```diff
@@ -0,0 +1,146 @@
+import isEmpty from "lodash.isempty";
+import { ARTICLE_REACTION, COMMENT_REACTION } from "@helpers/constants";
+import { typeOfReaction } from "@helpers/reaction";
+
+const { Reactions } = models;
+
+export const likeOrDislikeAnArticle = async (req, res) => {
+  const {
+    article,
+    user: { userId },
+    body: { reaction },
+    article: { title }
+  } = req;
+  try {
+    const articleReaction = await Reactions.findOne({
+      where: {
+        articleId: article.id,
+        userId
+      }
+    });
+
+    if (isEmpty(articleReaction)) {
+      await Reactions.create({
+        articleId: article.id,
+        sourceType: ARTICLE_REACTION,
+        reaction,
+        userId
+      });
+      await article.increment(`${typeOfReaction(reaction).count}`);
+      return httpResponse(res, {
+        statusCode: 201,
+        message: `You have successfully added a ${
+          typeOfReaction(reaction).type
+        } reaction to ${title}`
+      });
+    }
+    await Reactions.destroy({
+      where: {
+        id: articleReaction.id
+      }
+    });
+    await article.decrement(
+      `${typeOfReaction(articleReaction.get("reaction")).count}`
+    );
+    return httpResponse(res, {
+      statusCode: 200,
+      message: `You have successfully removed your ${
+        typeOfReaction(articleReaction.get("reaction")).type
+      } reaction to ${title}`
+    });
+  } catch (error) {
+    return serverError(res, error);
+  }
+};
+export const likeOrDislikeACommment = async (req, res) => {
+  const {
+    comment,
+    user: { userId },
+    body: { reaction },
+    comment: { content }
+  } = req;
+  try {
+    const commentReaction = await Reactions.findOne({
+      where: {
+        commentId: comment.id,
+        userId
+      }
+    });
+
+    if (isEmpty(commentReaction)) {
+      await Reactions.create({
+        commentId: comment.id,
+        sourceType: COMMENT_REACTION,
+        reaction,
+        userId
+      });
+      await comment.increment(`${typeOfReaction(reaction).count}`);
+      return httpResponse(res, {
+        statusCode: 201,
+        message: `You have successfully added a ${
+          typeOfReaction(reaction).type
+        } reaction to ${content}`
+      });
+    }
+    await Reactions.destroy({
+      where: {
+        id: commentReaction.id
+      }
+    });
+    await comment.decrement(
+      `${typeOfReaction(commentReaction.get("reaction")).count}`
+    );
+    return httpResponse(res, {
+      statusCode: 200,
+      message: `You have successfully removed your ${
+        typeOfReaction(commentReaction.get("reaction")).type
+      } reaction to ${content}`
+    });
+  } catch (error) {
+    return serverError(res, error);
+  }
+};
+export const fetchAllArticleReactions = async (req, res) => {
+  const {
+    article: { id, title }
+  } = req;
+  const articleReactions = await Reactions.findAll({
+    where: {
+      articleId: id
+    }
+  });
+  if (isEmpty(articleReactions)) {
```

Comment:
Please do not just call lodash everytime you need to check. I only installed as a last result